Add a webapp directory that contains the webapp for the config designer

CatFish47 opened the pull request.

CatFish47 committed:
Initialized react project

CatFish47 committed:
Added MUI and some components

CatFish47 committed:
added react-redux

CatFish47 committed:
added redux states for config and selected'

CatFish47 committed:
added more modules to body

CatFish47 committed:
fixed app start bug

CatFish47 committed:
Added param editing and config export

jenetic committed:
Display name of selected model on sidebar component when chosen

jenetic committed:
Change sidebar component color to yellow ('in progress') when user chooses model

jenetic committed:
Merge branch 'main' into sidebar-state
merge with main

jenetic committed:
Change sidebar color to green if user filled out all parameters

jenetic committed:
Clean up sidebar color changer code a bit

jenetic committed:
Remove param from newConfig if user deletes entry from param text box to maintain proper param count

CatFish47 committed:
Merge pull request #1 from CatFish47/sidebar-state

Make sidebar component change color based on status (not started, in progress, done) & display selected model

CatFish47 committed:
Fixed sidebar styling

jenetic committed:
Add button to download as JSON

jenetic committed:
Fix styling for download json button

jenetic committed:
Remove unnecessary function call

CatFish47 committed:
Fixed styling for modal buttons

CatFish47 committed:
Merge pull request #2 from CatFish47/download-json-button

Create button to download config as JSON file

CatFish47 committed:
add util files for type checking and html parsing

justyce22 committed:
Update utils.js

kathyychenn committed:
Create .DS_Store

kathyychenn committed:
Merge branch 'main' of https://github.com/CatFish47/FAVITES-Lite

kathyychenn committed:
Update utils.js

CatFish47 committed:
Add status changes

CatFish47 committed:
Merge branch 'main' of https://github.com/CatFish47/FAVITES-Lite

CatFish47 committed:
Made changes to type checking

kathyychenn committed:
installl

jenetic committed:
Add empty param object to each model, turn strings to number types when user saves config

kathyychenn committed:
Update utils.js

finish CSV, function and invalid param tests

kathyychenn committed:
update utils.js

fix errors in CSV test

CatFish47 committed:
Merge branch 'niemasd:main' into main

CatFish47 committed:
Fix header for laptop screen sizes

CatFish47 committed:
fix comma separate list typing

CatFish47 committed:
Merge branch 'main' into jenny-bug-fixes-2

CatFish47 committed:
Merge pull request #3 from CatFish47/jenny-bug-fixes-2

Bug fixes

CatFish47 committed:
Merge branch 'main' into kathy-param-tests

CatFish47 committed:
Merge pull request #4 from CatFish47/kathy-param-tests

param checks

CatFish47 committed:
fixed bug, added favicon and changed title

CatFish47 committed:
Merge branch 'main' of https://github.com/CatFish47/FAVITES-Lite

niemasd commented:
Thanks! A couple of comments before merging:

* Can you delete `.DS_Store` and add it to the `.gitignore` of the repo?
* What is `package-lock.json`? Does it need to be in the repo, or should it be deleted and added to the `.gitignore`?

CatFish47 committed:
Add DS Store to gitignore

CatFish47 committed:
Delete .DS_Store

CatFish47 commented:
> * Can you delete `.DS_Store` and add it to the `.gitignore` of the repo?

Sounds good! I just deleted `.DS_Store` and added it to `.gitignore`.

> * What is `package-lock.json`? Does it need to be in the repo, or should it be deleted and added to the `.gitignore`?

`package-lock.json` is a file that contains all of the webapp's package information so that when someone clones the repo they can easily install all packages and dependencies required to run the webapp. This is intended to be in the repo
